### PR TITLE
Add contract calls to admin page

### PR DESCRIPTION
**Pull request by marshall2112**
# Description

Hooking up contract calls with loading, error states in the UI.

- Create Pool
- updateWeightsGradually
- setSwapEnabled

Still pending (Future PR):
- Drain

# Checklist
- [ ] Code follows the style guide
- [ ] I have performed a self-review of my own code
- [ ] New and existing tests pass locally
- [ ] This PR is targeting the correct branch

### CODE REVIEW

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/use-pool-factory.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  const [ownerAddress, setOwnerAddress] = useState<string>();
+
+  useEffect(() => {
+    if (!signer) {
+      return;
+    }
+
+    (async () => {
+      const ownerAddress = await signer.getAddress();
```
Comment (mrfujisawa) [inline apps/dapp/src/components/Pages/Ascend/components/LBPForm/use-pool-factory.ts:29]: should be able to just do `const { wallet } = useWallet()`  to get the user address

Comment (marshall2112) [reply apps/dapp/src/components/Pages/Ascend/components/LBPForm/use-pool-factory.ts]: https://github.com/TempleDAO/temple/pull/461/commits/d6d1fcbad30026cafa8551085e227526b99551e5

File: apps/dapp/src/components/Layouts/Ascend/utils.ts [hunk 2]
```diff
@@ -24,12 +26,12 @@ export const createPool = (pool: SubgraphPool): Pool => {
       endTimestamp: new Date(Number(weightUpdate.endTimestamp) * 1000),
       startTimestamp: new Date(Number(weightUpdate.startTimestamp) * 1000),
       startWeights: [
-        getBigNumberFromString(weightUpdate.startWeights[0]),
-        getBigNumberFromString(weightUpdate.startWeights[1]),
+        BigNumber.from(weightUpdate.startWeights[0]),
```
Comment (mrfujisawa) [inline apps/dapp/src/components/Layouts/Ascend/utils.ts:29]: We should probably just make these DecimalBigNumbers from the start?

Comment (mrfujisawa) [reply apps/dapp/src/components/Layouts/Ascend/utils.ts:29]: if it changes too much downstream we can do a followup/cleanup after we merge this/higher priority stuff

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx [hunk 3]
```diff
@@ -1,114 +1,81 @@
...
       releasedToken: env.tokens.temple.address,
       accruedToken: env.tokens.frax.address,
-      startWeight: [BigNumber.from(0), BigNumber.from(0)],
+      startWeight1: DecimalBigNumber.fromBN(ZERO, 18),
+      startWeight2: DecimalBigNumber.fromBN(ZERO, 18),
       startDate: new Date(0),
       endDate: new Date(0),
       fees: 1,
       address: '',
+      endWeight1: DecimalBigNumber.fromBN(ZERO, 18),
```
Comment (mrfujisawa) [inline apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:43]: for these ZERO weights, there is a `DBN_ZERO` constant exported from the DecimalBigNumber file.  should probably just use the constant for all these 0 values?

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx [hunk 4]
```diff
@@ -119,73 +86,133 @@ export const LBPForm = ({ pool }: Props) => {
   };
 
   const createChangeHandler = (fieldKey: keyof Values) => (event: React.ChangeEvent<any>) => {
-    setFormValue(fieldKey, event.target.value);
+    let value = event.target.value;
+
+    if (['startWeight1', 'startWeight2', 'endWeight1', 'endWeight2'].indexOf(fieldKey) >= 0) {
+      value = DecimalBigNumber.parseUnits(value, 16);
```
Comment (mrfujisawa) [inline apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:92]: there are 16 decimals on these numbers? not 18?

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx [hunk 5]
```diff
@@ -119,73 +86,133 @@ export const LBPForm = ({ pool }: Props) => {
   };
 
   const createChangeHandler = (fieldKey: keyof Values) => (event: React.ChangeEvent<any>) => {
-    setFormValue(fieldKey, event.target.value);
+    let value = event.target.value;
+
+    if (['startWeight1', 'startWeight2', 'endWeight1', 'endWeight2'].indexOf(fieldKey) >= 0) {
```
Comment (mrfujisawa) [inline apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:91]: prob more appropriate to just do 
`['startWeight1', 'startWeight2', 'endWeight1', 'endWeight2'].includes(fieldKey)`

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx [hunk 6]
```diff
@@ -119,73 +86,133 @@ export const LBPForm = ({ pool }: Props) => {
...
+      value = DecimalBigNumber.parseUnits(value, 16);
+    }
+
+    if (['startDate', 'endDate'].indexOf(fieldKey) >= 0) {
```
Comment (mrfujisawa) [inline apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:95]: `.includes`

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx [hunk 7]
```diff
@@ -119,73 +86,133 @@ export const LBPForm = ({ pool }: Props) => {
...
+    } catch (error) {
+      console.error(error);
+      result = format(new Date(), "yyyy-LL-dd'T'kk:mm");
+    }
+    return result;
+  };
+
+  const formatWeight = (weight: DecimalBigNumber) => {
+    if (!weight) return '0';
+    return Math.trunc(Number(formatUnits(weight.value, 16)));
```
Comment (mrfujisawa) [inline apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:150]: DecimalBigNumber has the `formatUnits` as part of the class -- can just do 
```
Math.trunc(Number(weight.formatUnits()))
```

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx [hunk 8]
```diff
@@ -119,73 +86,133 @@ export const LBPForm = ({ pool }: Props) => {
...
-            <option value={token.address} key={token.address}>
-              {token.name}
-            </option>
-          ))}
-        </Select>
+        {isEditMode ? (
+          <h3>{pool.tokens[0].symbol}</h3>
+        ) : (
+          <Select id="released-token" onChange={createChangeHandler('releasedToken')} value={formValues.releasedToken}>
+            {Object.values(env.tokens).map((token) => (
```
Comment (mrfujisawa) [inline apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:192]: we'll need to clean up the `env.tokens` as a follow up here.  We should probably validate what tokens we care about here and make sure the env values are correct

Comment (mrfujisawa) [reply apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:192]: doesnt need to be part of this PR

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/use-pool-contract.ts [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    setIsSetSwapEnabledLoading(true);
+    setSetSwapEnabledError(null);
+    try {
+      const result = await poolContract!.setSwapEnabled(enabled, {
+        gasLimit: 400000,
+      });
+      await result.wait();
+    } catch (error: any) {
+      console.error(error);
+      setSetSwapEnabledError('Transaction failed.');
```
Comment (mrfujisawa) [inline apps/dapp/src/components/Pages/Ascend/components/LBPForm/use-pool-contract.ts:49]: if it made sense (not sure if it does) could use the NotificationManager context and just show notifications on create/error.

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx (hunk 4 above)
Comment (marshall2112) [reply apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:92]: The issue is the value from the input is already taking up 2 decimal places, e.g. 80 vs 0.8. So we get two decimal places "for free".

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx (hunk 3 above)
Comment (marshall2112) [reply apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:43]: Agreed 👍

File: apps/dapp/src/components/Layouts/Ascend/utils.ts (hunk 2 above)
Comment (marshall2112) [reply apps/dapp/src/components/Layouts/Ascend/utils.ts:29]: Yes good call, seems like an easy change will do it.

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx (hunk 7 above)
Comment (marshall2112) [reply apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:150]: I'm not sure why but that isn't giving me the expected result. Unless you strongly object we can leave it as is and I can work on it in another PR.

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx (hunk 8 above)
Comment (marshall2112) [reply apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx:192]: Yes was thinking that could be a follow up PR as well. I will clarify in the channel and get that going.

File: apps/dapp/src/components/Layouts/Ascend/utils.ts (hunk 2 above)
Comment (marshall2112) [reply apps/dapp/src/components/Layouts/Ascend/utils.ts]: https://github.com/TempleDAO/temple/pull/461/commits/bb507608688ba733e68f2319522ec1215bc9784f

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx (hunk 3 above)
Comment (marshall2112) [reply apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx]: https://github.com/TempleDAO/temple/pull/461/commits/bb507608688ba733e68f2319522ec1215bc9784f

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx (hunk 5 above)
Comment (marshall2112) [reply apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx]: https://github.com/TempleDAO/temple/pull/461/commits/bb507608688ba733e68f2319522ec1215bc9784f

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx (hunk 6 above)
Comment (marshall2112) [reply apps/dapp/src/components/Pages/Ascend/components/LBPForm/index.tsx]: https://github.com/TempleDAO/temple/pull/461/commits/bb507608688ba733e68f2319522ec1215bc9784f

File: apps/dapp/src/components/Pages/Ascend/components/LBPForm/use-pool-contract.ts (hunk 9 above)
Comment (marshall2112) [reply apps/dapp/src/components/Pages/Ascend/components/LBPForm/use-pool-contract.ts]: https://github.com/TempleDAO/temple/pull/461/commits/bb507608688ba733e68f2319522ec1215bc9784f

**Review by mrfujisawa — APPROVED** [review]

**Pull request merged**